configs/11.0/packages/gcc/stage_2

```diff
@@ -109,7 +110,8 @@ atcfg_configure() {
 			--with-mpc-include="${at_dest}/include" \
 			--with-mpc-lib=${libdir} \
 			--with-as="${at_dest}/bin/as" \
-			--with-ld="${at_dest}/bin/ld"
+			--with-ld="${at_dest}/bin/ld" \
+			--with-stage1-ldflags="-Wl,--dynamic-linker=${at_dest}/lib64/ld64.so.2"
```

Comment:
This line is the trick one and that I am not sure is the right approach.

By setting CC/CXX it fix all search paths for include and linking while building gcc_2 and gcc_3.
But It does not fix the library search path for execution. This is clear during gcc_2 compilation. It compiles a helper program called genautomata and tries to run it as parte of the make script. It is built against glibc from AT but it is run with the host loader (/lib64/ld.so.2) which uses the host library paths to search libs. In the case of AT13 and ubuntu 18.04 it fails because the host AT is too old for the newly build program, and the loader complains.

There is some ways to fix it. The ones I though of are:
a) Use rpath to hardcode runtime library paths for built binaries.
b) Use LD_LIBRARY_PATH to use AT libraries.
c) Change the `.interp` loader in the elf of built binaries for the AT ld.so.2 that is already configured with the right search paths.

I choose (c). It seems that gcc by default (or ld) uses the same loader as itself for the elfs it build (That is why genautomata built with gcc_1 have the host loader in `.interp`). But after creating glibc_1 we can start using the new loader. And once you change it for gcc_2 it becomes the default loader for all elfs built with it for now on and later gcc_3 and so on.

Reply:
I'm ok with your solution.
We set LIBRARY_PATH in the stage 4 for AT 9.0, you could do the same to have a sort of consistency if you don't need to set it again on the later stages. I wasn't expect this behaviour to happen since you are setting CC/CXX to AT.

Reply:
>By setting CC/CXX it fix all search paths for include and linking while building gcc_2 and gcc_3.
But It does not fix the library search path for execution. This is clear during gcc_2 compilation. It compiles a helper program called genautomata and tries to run it as part of the make script. It is built against glibc from AT but it is run with the host loader (/lib64/ld.so.2) which uses the host library paths to search libs. In the case of AT13 and ubuntu 18.04 it fails because the host glibc is too old for the newly build program, and the loader complains.

I think you should add this paragraph **as is** to the commit message.  And I suggest that you add it as a comment in the code, with a little rephrasing of the first 3 sentences so that they don't describe what's changed, but what's accomplished with the code.  Suggestion:

_The use of --with-stage1-ldflags is required during gcc_2 compilation. gcc_2 compiles a helper program called genautomata and tries to run it as part of the make script. It is correctly built against glibc from AT, but it is run with the host loader (/lib64/ld.so.2), which uses the host library paths to search libs, potentially leading to build failures._

(as a comment in the code).

Reply:
@rff Oops.  The loader path is correct correct for ppc64le (ELFv2), but is wrong for ppc and ppc64.  In those cases they should point to:
- ppc: `lib/ld.so.1`
- ppc64: `lib64/ld64.so.1`